service/internal/idpplugin/keycloak_plugin.go

```diff
@@ -94,95 +127,90 @@ func EntityResolution(ctx context.Context,
-					connector.token.AccessToken,
-					kcConfig.Realm,
-					gocloak.GetGroupsParams{Search: func() *string { t := ident.GetEmailAddress(); return &t }()},
-				)
-				if groupErr != nil {
-					slog.Error("Error getting group", "group", groupErr)
-					return &authorization.IdpPluginResponse{},
-						status.Error(codes.Internal, db.ErrTextGetRetrievalFailed)
-				} else if len(groups) == 1 {
-					slog.Info("Group found for", "entity", ident.String())
-					group := groups[0]
-					expandedRepresentations, exErr := expandGroup(*group.ID, connector, &kcConfig, ctx)
-					if exErr != nil {
+		} else if getUserParams != (gocloak.GetUsersParams{}) {
+			// if getUserParams not empty
+			users, err := connector.client.GetUsers(ctx, connector.token.AccessToken, kcConfig.Realm, getUserParams)
+			if err != nil {
+				slog.Error(err.Error())
+				return &authorization.IdpPluginResponse{},
+					status.Error(codes.Internal, db.ErrTextGetRetrievalFailed)
+			}
+			if len(users) == 1 {
+				user := users[0]
+				slog.Debug("User found", "user", *user.ID, "entity", ident.String())
+				slog.Debug("User", "details", fmt.Sprintf("%+v", user))
+				slog.Debug("User", "attributes", fmt.Sprintf("%+v", user.Attributes))
+				keycloakEntities = append(keycloakEntities, user)
+			} else {
+				slog.Error("No user found for", "entity", ident.String())
+				if ident.GetEmailAddress() != "" {
+					// try by group
+					groups, groupErr := connector.client.GetGroups(
+						ctx,
+						connector.token.AccessToken,
+						kcConfig.Realm,
+						gocloak.GetGroupsParams{Search: func() *string { t := ident.GetEmailAddress(); return &t }()},
+					)
+					if groupErr != nil {
+						slog.Error("Error getting group", "group", groupErr)
 						return &authorization.IdpPluginResponse{},
-							status.Error(codes.Internal, db.ErrTextNotFound)
-					} else {
-						keycloakEntities = expandedRepresentations
+							status.Error(codes.Internal, db.ErrTextGetRetrievalFailed)
 					}
-				} else {
-					slog.Error("No group found for", "entity", ident.String())
-					var entityNotFoundErr authorization.EntityNotFoundError
-					switch ident.GetEntityType().(type) {
-					case *authorization.Entity_EmailAddress:
-						entityNotFoundErr = authorization.EntityNotFoundError{Code: int32(codes.NotFound), Message: db.ErrTextGetRetrievalFailed, Entity: ident.GetEmailAddress()}
-					case *authorization.Entity_UserName:
-						entityNotFoundErr = authorization.EntityNotFoundError{Code: int32(codes.NotFound), Message: db.ErrTextGetRetrievalFailed, Entity: ident.GetUserName()}
-					// case "":
-					// 	return &authorization.IdpPluginResponse{},
-					// 		status.Error(codes.InvalidArgument, db.ErrTextNotFound)
-					default:
-						slog.Error("Unsupported/unknown type for", "entity", ident.String())
-						entityNotFoundErr = authorization.EntityNotFoundError{Code: int32(codes.NotFound), Message: db.ErrTextGetRetrievalFailed, Entity: ident.String()}
+					if len(groups) == 1 {
```

Comment:
Same here is it possible to get more than one group?

Reply:
it is possible, let's modify the `slog.Error` to "Unexpected group count found for "entity", ident.String(), "count",  len(groups)"